apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,669 @@
+- `message` The message to show the user if the decision was `reject`.
+- `should_logout_user` Whether to logout a user if a `reject` decision is issued. Has no effect when a `continue` decision is issued.
+
+Example output:
+
+```json
+{
+   "decision": "reject",
+   "message" : "You have exceeded maximum number of MFA attempts.",
+   "should_logout_user": "false"
+}
+```
+
+****************On Error****************
+
+Return this if your hook processed the input with some error.
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Must be set.
+- In general all other fields attached to the returned JSON are ignored.
+
+Example output:
+
+```jsx
+{
+  "error": {
+    "http_code": 429,
+    "message": "You can only verify a factor once very 10 seconds."
+  }
+}
+```
+
+## Custom Access Token Hook
+
+You may wish to add additional claims based on the authentication method used or add additional metadata to select users. The custom access token hook runs before a token is issued and allows you to add claims.
+
+### Inputs
+
+- `user_id`Unique identifier for the user attempting to sign in. Correlate this to the `auth.users` table.
+- `claims` Claims which are attached to the access token.
+- `authentication_method` the authentication method used to request for the access token. Possible values include:
+    - `oauth`
+    - `password`
+    - `otp`
+    - `totp`
+    - `recovery`
+    - `invite`
+    - `sso/saml`
+    - `magiclink`
+    - `email/signup`
+    - `email_change`
+    - `token_refresh`
+
+### Outputs
+
+In exchange your hook should return back a JSON object with these fields set:
+
+********************************On Success********************************
+
+Return these only if your hook processed the input without errors.
+
+- `claims` A json containing the updated claims after the hook has been run
+
+****************On Error****************
+
+Return this if your hook processed the input with some error.
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Must be set.
+- In general all other fields attached to the returned JSON are ignored.
+
+
+
+<Admonition type="warn">
+Claims returned must conform to our specification. While you are free to add claims, you can only remove the `user_metadata` and app_metadata claims as we rely on the rest  of the claims for proper functioning of the service.
+</Admonition>
```

Comment:
This shouldn't be an admonition. It should be explained properly at the top of the hook's description. Please add the list of claims to the User Sessions page and link to them here.

Reply:
we should also call out the claims that can't be removed (prevented by us) like the `role` , `exp`, etc...